src/Microsoft.DotNet.Build.CloudTestTasks/AzureHelper.cs

```diff
@@ -0,0 +1,234 @@
+            {
+                signedpermissions += "wdl";
+            }
+            string signedstart = DateTime.UtcNow.ToString("O");
+            string signedexpiry = DateTime.UtcNow.AddDays(tokenValue).ToString("O");
+            string canonicalizedresource = "/blob/" + AccountName + "/" + ContainerName;
+            string signedidentifier = "";
+            string signedversion = "2015-04-05";
+            string rscc = "";
+            string rscd = "";
+            string rsce = "";
+            string rscl = "";
+            string rsct = "";
+
+
+            string StringToSign = signedpermissions + "\n" +
+               signedstart + "\n" +
+               signedexpiry + "\n" +
+               canonicalizedresource + "\n" +
+               signedidentifier + "\n" +
+               "" + "\n" +
+               "" + "\n" +
+               signedversion + "\n" +
+               rscc + "\n" +
+               rscd + "\n" +
+               rsce + "\n" +
+               rscl + "\n" +
+               rsct;
+
+            byte[] SignatureBytes = System.Text.Encoding.UTF8.GetBytes(StringToSign);
+            System.Security.Cryptography.HMACSHA256 SHA256 = new System.Security.Cryptography.HMACSHA256(Convert.FromBase64String(key));
+            string sig = Convert.ToBase64String(SHA256.ComputeHash(SignatureBytes));
+
+            string sasURL = string.Format("?sv={0}&sr={1}&sig={2}&st={3}&se={4}&sp={5}",
+                WebUtility.UrlEncode(signedversion),
+                WebUtility.UrlEncode("c"),
+                WebUtility.UrlEncode(sig),
+                WebUtility.UrlEncode(signedstart),
+                WebUtility.UrlEncode(signedexpiry),
+                WebUtility.UrlEncode(signedpermissions)
+                );
+
+            return sasURL;
+        }
+
+
+        public static string AuthorizationHeader(string StorageAccount, string StorageKey, string method, DateTime now,
+            HttpWebRequest request, string ifMatch = "", string md5 = "", string size="",string contentType="")
+        {
+            var MessageSignature = string.Format("{0}\n\n\n{1}\n{5}\n{6}\n\n\n{2}\n\n\n\n{3}{4}",
+                method,
+                (size == String.Empty) ? string.Empty : size,
+                ifMatch,
+                GetCanonicalizedHeaders(request),
+                GetCanonicalizedResource(request.RequestUri, StorageAccount),
+                md5,
+                contentType
+                );
+
+            var SignatureBytes = Encoding.UTF8.GetBytes(MessageSignature);
+            var SHA256 = new HMACSHA256(Convert.FromBase64String(StorageKey));
+            var AuthorizationHeader = "SharedKey " + StorageAccount + ":" +
+                                      Convert.ToBase64String(SHA256.ComputeHash(SignatureBytes));
+            return AuthorizationHeader;
+        }
+
+        public static string GetCanonicalizedHeaders(HttpWebRequest request)
+        {
+            var headerNameList = new ArrayList();
```

Comment:
ArrayList is bad, especially when you are force casting things in it to strings. Use List<string>